docs/source/docs/configuration.md

```diff
@@ -2,19 +2,459 @@
 
 Besides the arguments provided to the command line tools, detailed training and
 prediction configuration is specified in a `yaml` file. The configuration file contains
-two sections, `dataset` and `train`.
+two sections, `dataset` and `train`. Within `dataset` one specifies the data types,
+sizes, as well as the data loader to use. The `train` section specifies parameters
+related to the neural network. `yaml` files are used as they can be easily generated
+from basic Python dictionary structures.
 
 ## Dataset section
 
+The `dataset` section specifies the path to the data to be used during training, the
+data loader to use as well as the specific arguments to configure the data loader.
+
+### Dir key - Required
+
+The paths to the training, validation and testing data are specified under a `dir`
+dictionary key like this:
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train" # folder contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+```
+
+Multiple dataset directories can be specified, such that data is sampled across several
+folders:
+
+```yaml
+dataset:
+  dir:
+    train:
+      - "data/test/h5/paired/train1"
+      - "data/test/h5/paired/train2" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
```

Comment:
do you think we should add other required args in these previous examples as well?

Reply:
So my idea was to give examples on "how to build it" bit by bit, so that the different parts don't get confused with each other. Is it worth adding a sentence about this to clarify?